ui/app/components/pages/confirm-transaction-base/confirm-transaction-base.container.js

```diff
@@ -48,7 +57,10 @@ const mapStateToProps = (state, props) => {
   const { balance } = accounts[selectedAddress]
   const { name: fromName } = identities[selectedAddress]
   const toAddress = propsToAddress || txParamsToAddress
-  const toName = identities[toAddress] && identities[toAddress].name
+  const toName = identities[toAddress]
+    ? identities[toAddress].name
+    : casedContractMap[toAddress] ? casedContractMap[toAddress].name : addressSlicer(toAddress)
```

Comment:
> > Will that be a good solution?
>
> Showing the sliced address is a good fallback, but in this case, [the StandardBounties contract name is in our eth-contract-metadata repo](https://github.com/MetaMask/eth-contract-metadata/blob/master/contract-map.json#L527), so we should default to that `.name` property first.

— [Comment on #4836](https://github.com/MetaMask/metamask-extension/issues/4836#issuecomment-406680012)

If I understand Finlay's comment, we should have this the other way around:

```js
const toName = casedContractMap[toAddress]
  ? casedContractMap[toAddress].name
  : (
    identities[toAddress]
      ? identities[toAddress].name
      : addressSlicer(toAddress)
    )
```

Reply:
I assumed he was talking about the eth-contract-metadata contract map having priority over the sliced address. `identities` should just be user-held accounts and probably wouldn't have very many values.

Reply:
Ah, fair enough